src/Processors/Transforms/MongoDBSource.cpp

```diff
@@ -341,17 +342,20 @@ Chunk MongoDBSource::generate()
             for (const auto idx : collections::range(0, size))
             {
                 const auto & name = description.sample_block.getByPosition(idx).name;
+                bool is_nullable = description.types[idx].second;
 
-                if (strict_check_names && !document->exists(name))
+                if (!is_nullable && strict_check_names && !document->exists(name))
```

Comment:
Why not to allow inserting defaults for missing non-nullable columns?

Reply:
makes sense